game/quake2_md2.ksy

```diff
@@ -0,0 +1,433 @@
+  ```
+
+  The above are filled in for player models; for the separate weapon models,
+  the final frame is 173 "g_view" (unknown purpose) since weapons aren't shown
+  during death animations. `a_grenades.md2`, the handgrenade weapon model, is
+  the same except that the `wave` frames are blank (according to the default
+  female model files). This is likely due to its dual use as a grenade throw
+  animation where this model must leave the player's model.
+doc-ref:
+  - https://icculus.org/~phaethon/q3a/formats/md2-schoenblum.html
+  - http://tfc.duke.free.fr/coding/md2-specs-en.html
+  - http://tastyspleen.net/~panjoo/downloads/quake2_model_frames.html
+  - http://wiki.polycount.com/wiki/OldSiteResourcesQuake2FramesList
+seq:
+  - id: magic
+    contents: IDP2
+  - id: version
+    type: u4
+    valid: 8
+  - id: skin_width_px
+    type: u4
+  - id: skin_height_px
+    type: u4
+  - id: bytes_per_frame
+    type: u4
+  - id: num_skins
+    type: u4
+  - id: vertices_per_frame
+    type: u4
+  - id: num_tex_coords
+    type: u4
+  - id: num_triangles
+    type: u4
+  - id: num_gl_cmds
+    type: u4
+  - id: num_frames
+    type: u4
+  - id: ofs_skins
+    type: u4
+  - id: ofs_tex_coords
+    type: u4
+  - id: ofs_triangles
+    type: u4
+  - id: ofs_frames
+    type: u4
+  - id: ofs_gl_cmds
+    type: u4
+  - id: ofs_eof
+    type: u4
+instances:
+  skins:
+    pos: ofs_skins
+    size: 64
+    type: strz
+    encoding: ascii
+    repeat: expr
+    repeat-expr: num_skins
+  tex_coords:
+    pos: ofs_tex_coords
+    type: tex_point
+    repeat: expr
+    repeat-expr: num_tex_coords
+  triangles:
+    pos: ofs_triangles
+    type: triangle
+    repeat: expr
+    repeat-expr: num_triangles
+  frames:
+    pos: ofs_frames
+    size: bytes_per_frame
+    type: frame
+    repeat: expr
+    repeat-expr: num_frames
+  gl_cmds:
+    pos: ofs_gl_cmds
+    # in http://tfc.duke.free.fr/coding/src/md2.c initially read as `int *`
+    # (a pool of `int`s) => hence `s4`
+    size: 4 * num_gl_cmds # FIXME: replace `4` with `sizeof<s4>` once `sizeof` is implemented for GraphViz
+    type: gl_cmds_list
+  anim_names:
+    value: |
+      ['stand', 'run', 'attack', 'pain1', 'pain2', 'pain3', 'jump', 'flip',
+      'salute', 'taunt', 'wave', 'point', 'crstnd', 'crwalk', 'crattak',
+      'crpain', 'crdeath', 'death1', 'death2', 'death3']
+  anim_start_indices:
+    value: |
+      [0, 40, 46, 54, 58, 62, 66, 72,
+      84, 95, 112, 123, 135, 154, 160,
+      169, 173, 178, 184, 190]
+  anim_num_frames:
+    value: |
+      [40, 6, 8, 4, 4, 4, 6, 12,
+      11, 17, 11, 12, 19, 6, 9,
+      4, 5, 6, 6, 8]
+  anorms_table:
+    doc-ref: |
+      https://github.com/skullernet/q2pro/blob/f4faabd/src/common/math.c#L80
+      from Quake anorms.h
+    value: |
+      [
+        [-0.525731, 0.000000, 0.850651], 
+        [-0.442863, 0.238856, 0.864188], 
+        [-0.295242, 0.000000, 0.955423], 
```

Comment:
There should be no trailing whitespace (on _any_ line; I've only highlighted some of them). It's recommended to set up your editor to honor the [project `.editorconfig`](https://github.com/kaitai-io/kaitai_struct_formats/blob/1eae5145325bb992e20dae838dd68d04814dda46/.editorconfig#L8) (see https://editorconfig.org/ for explanation).

If you use VS Code, it doesn't support EditorConfig natively, but there is an official [_EditorConfig_](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) extension; for other editors, see https://editorconfig.org/#download.

Reply:
Cool, wasn't aware of that. I'm actually using the GitHub web editor for these small changes.